### ISSUE TITLE
Generate an images sources text file

### ISSUE THREAD

**Issue by rmweir** [rancher/rancher#31663]
**Description:**
Image export script should generate a file similar to rancher-images.txt that includes sources (what uses the image in rancher). Source can be "system" meaning it is for RKE or needed for base functionality such as pipelines, "core" meaning it is needed for core rancher functionality, "rancher" meaning it is the rancher or rancher-agent image, "k3sUpgrade", or any of the charts/system-charts.

Testing:
To test compare rancher-images.txt to previous version and make sure there aren't any unexpected difference. An unexpected difference would be something like many images missing. Another thing to validate is the the rancher-images-sources.txt should contain the same number of rows as the rancher-images.txt.

**Related Issue:**
https://github.com/rancher/rancher/issues/29893

**Comment by aaronRancher** [comment]
Tested by comparing rancher-images.txt from `v2.5.7` to `v2.5.8-rc2`

Saw the following images in `v2.5.7` were not present in `v2.5.8-rc2`:
```
rancher/externalip-webhook:v0.1.6
rancher/istio-installer:1.8.3-rancher1	
rancher/mirrored-cloud-provider-vsphere-cpi-release-manager:v1.2.1	
rancher/mirrored-cloud-provider-vsphere-csi-release-driver:v2.1.0	
rancher/mirrored-cloud-provider-vsphere-csi-release-syncer:v2.1.0
rancher/rancher-webhook:v0.1.0-beta9	
```

**Comment by aaronRancher** [comment]
Validated `rancher-images-sources.txt` have the same number of rows as `rancher-images.txt`.

**Comment by cbron** [comment]
@aaronRancher lets spin out those 6 images to a new bug issue, its because no new charts have been added to dev for those charts

**Comment by aaronRancher** [comment]
Found additional images included in `v2.5.7` and missing from `v2.5.8-rc2` `rancher-images.txt`
```
rancher/mirrored-coreos-kube-state-metrics:v1.9.7
rancher/mirrored-coreos-prometheus-config-reloader:v0.38.1
rancher/mirrored-coreos-prometheus-operator:v0.38.1
rancher/mirrored-istio-coredns-plugin:0.2-istio-1.1	
rancher/mirrored-istio-install-cni:1.8.3	
rancher/mirrored-istio-pilot:1.8.3	
rancher/mirrored-istio-proxyv2:1.8.3	
rancher/mirrored-jaegertracing-all-in-one:1.20.0
rancher/mirrored-k8scsi-csi-attacher:v3.0.0	
rancher/mirrored-k8scsi-csi-node-driver-registrar:v2.0.1	
rancher/mirrored-k8scsi-csi-provisioner:v2.0.0	
rancher/mirrored-k8scsi-csi-resizer:v1.0.0	
rancher/mirrored-k8scsi-livenessprobe:v2.1.0
rancher/mirrored-kiali-kiali:v1.29.0
rancher/mirrored-kube-rbac-proxy:v0.5.0
rancher/mirrored-longhornio-csi-attacher:v2.2.1-lh1	
rancher/mirrored-longhornio-csi-node-driver-registrar:v1.2.0-lh1	
rancher/mirrored-longhornio-csi-provisioner:v1.6.0-lh1	
rancher/mirrored-longhornio-csi-resizer:v0.5.1-lh1	
rancher/mirrored-longhornio-csi-snapshotter:v2.1.1-lh1	
rancher/mirrored-longhornio-longhorn-engine:v1.1.0	
rancher/mirrored-longhornio-longhorn-instance-manager:v1_20201216	
rancher/mirrored-longhornio-longhorn-manager:v1.1.0	
rancher/mirrored-longhornio-longhorn-share-manager:v1_20201204	
rancher/mirrored-longhornio-longhorn-ui:v1.1.0
rancher/mirrored-prom-node-exporter:v1.0.1
rancher/mirrored-prom-prometheus:v2.18.2
rancher/mirrored-squareup-ghostunnel:v1.5.2	
```

Related issue: https://github.com/rancher/rancher/issues/31995

**Comment by rmweir** [comment]
I'm checking each one of these, so far it looks like they have just beens lightly renamed or their version has been upgraded. I see `rancher/mirrored-istio-coredns-plugin`, `rancher/mirrored-jaegertracing-all-in-one`. `rancher/mirrored-k8scsi-csi-attacher`, `rancher/mirrored-k8scsi-csi-node-driver-registrar`, `rancher/mirrored-k8scsi-csi-provisioner`,  `rancher/mirrored-kube-rbac-proxy`, and all the longhorn imagges are probably related to the other issue. `rancher/mirrored-squareup-ghostunnel` is no longer used by monitoring v2, and these lists extract images from the latest version of each chart.
